src/Domain/Donation.php

```diff
@@ -819,43 +822,47 @@ public function getFundingWithdrawalTotalAsObject(): Money
     }
 
     /**
-     * @return string Total amount *finalised*, matched by `Fund`s of type "championFund"
-     */
-    private function getConfirmedChampionWithdrawalTotal(): string
-    {
-        if (!$this->getDonationStatus()->isSuccessful()) {
-            return '0.0';
-        }
-
-        $withdrawalTotal = '0.0';
-        foreach ($this->fundingWithdrawals as $fundingWithdrawal) {
-            if ($fundingWithdrawal->getCampaignFunding()->getFund()->getFundType() === FundType::ChampionFund) {
-                $withdrawalTotal = bcadd($withdrawalTotal, $fundingWithdrawal->getAmount(), 2);
-            }
-        }
-
-        return $withdrawalTotal;
-    }
-
-    /**
-     * @return string Total amount *finalised*, matched by `Fund`s of type "pledge"
+     * @return array{
+     *     amountMatchedByChampionFunds: numeric-string,
+     *     amountMatchedByPledges: numeric-string,
+     *     amountPreauthorizedFromChampionFunds: numeric-string,
+     *     amountMatchedOther: numeric-string,
+     * }
      */
-    private function getConfirmedPledgeWithdrawalTotal(): string
+    public function getWithdrawalTotalByFundType(): array
     {
-        if (!$this->getDonationStatus()->isSuccessful()) {
-            return '0.0';
-        }
+        $withdrawalTotals = [
+            'amountMatchedByChampionFunds' => '0.00',
+            'amountMatchedByPledges' => '0.00',
+            'amountPreauthorizedFromChampionFunds' => '0.00',
+            'amountMatchedOther' => '0.00', // This key is not sent to SF, covers match fund usage that we don't need to
+                                           // report, i.e. for donations that are neither sucessful nor preauthed.
+        ];
 
-        $withdrawalTotal = '0.0';
         foreach ($this->fundingWithdrawals as $fundingWithdrawal) {
-            $fundIsPledge = $fundingWithdrawal->getCampaignFunding()->getFund()->getFundType()->isAnyPledgeType();
-
-            if ($fundIsPledge) {
-                $withdrawalTotal = bcadd($withdrawalTotal, $fundingWithdrawal->getAmount(), 2);
-            }
+            $fundTypeOfThisWithdrawal = $fundingWithdrawal->getCampaignFunding()->getFund()->getFundType();
+
+            $key = match ([$fundTypeOfThisWithdrawal, $this->donationStatus->isSuccessful(), $this->donationStatus === DonationStatus::PreAuthorized  ]) {
```

Comment:
Took a moment to follow how this works at a high level but this does feel a lot more durable, looks good